### PR TITLE
simplify and focuse iam policies

### PR DESCRIPTION
**Pull request by tloubrieu-jpl**
## 🗒️ Summary

Updates needed while reviewing and tested the deployment in the Test venue with the SA's.

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [x] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [ ] AI generated substantial portions of this code

Estimated % of code influenced by AI: _50_ %

## ⚙️ Test Data and/or Report

Validated with automated integration tests in the Test venue.

## ♻️ Related Issues

Fixes: https://github.com/NASA-PDS/registry/issues/542

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### Terraform (only if this PR touches a `terraform/` directory)

- [ ] **Validator run:** `scripts/validate_terraform.py` was run against the changed `terraform/` tree and any Must-Have failures were resolved (or are tracked in a reviewed `.tfvalidate-ignore` entry, not silently bypassed).
- [ ] **Ownership boundary:** Shared/singleton CDS resources (Cognito, CloudFront, org-wide IAM roles, shared security groups) were only added/changed in `pdc-cds-infra`, not duplicated in an application repo.
- [ ] **Module structure:** `modules/` stays flat and no new module is just a thin wrapper around a single resource.
- [ ] **SSM interface:** Any new cross-component output is actually published under `/pds/<component>/...` and, if this PR is the consumer side, reads from the correct published parameter rather than another repo's Terraform state.
- [ ] **Naming intent:** Resource and variable names are semantically meaningful, not just non-redundant (the validator only checks for repeated resource-type substrings).
- [ ] **Auth at runtime:** Terraform/CI actually authenticates via OIDC or an assumed role when applied — not just "no static key found in this diff."
- [ ] **Least privilege:** Any new or changed IAM policy attached to the Terraform execution role grants only what this change needs.

### CODE REVIEW

File: scripts/manual-aoss-reindexer/reindex.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 # Baseline mappings which are required to facilitate successful execution before any data is copied
 necessary_mappings = {
     'lidvid': 'keyword',
-    'ops:Harvest_Info/ops:harvest_date_time': 'date'
+    'ops:Harvest_Info.ops:harvest_date_time': 'date'
```
Comment (jordanpadams) [inline scripts/manual-aoss-reindexer/reindex.py]: @tloubrieu-jpl I am renaming this directory if that is OK...

**Comment by jordanpadams** [comment]
@tloubrieu-jpl there are a lot of changes on this branch not related to IAM. is that intentional? are those changes coming from `develop`?

**Comment by tloubrieu-jpl** [comment]
Hi @jordanpadams , I was not expecting those changes. Something went unexpectingly with the branching, I converted the PR back to draft until I fix that.

**Comment by tloubrieu-jpl** [comment]
Yes, @jordanpadams , by mistake I branched the release-candidate branch off of develop. I will fix the PR now to remove the unwanted commit from develop.

**Comment by tloubrieu-jpl** [comment]
Actually I am seeing a faulty commit 933aaa81b700ac5d55ee7cae0aa68814e9e141d1

I will reset two commits before, we can move your commit c3726d30eb24c1644bb07d6e6a7ddd1424f4e5a1 to develop branch unless this is meant to be deployed in production shortly.

**Comment by tloubrieu-jpl** [comment]
It seems that I did a bad move for the github UI since the issue was not on my local repository. I reseted the branch 2 commits earlier and the changes are now what I expected.

**Comment by tloubrieu-jpl** [comment]
Hi @alexdunnjpl , could you review this PR ?
